### PR TITLE
Feat/faucet multi

### LINKED ISSUE #87: cmd/faucet: chain configurations
Currently the cmd/faucet doesn't have support for the many chains that cmd/geth does. 

I've made a quick `feat/faucet-kotti` branch that supports, well, Kotti, but it would be convenient and thorough-seeming to extend the multi-geth-standard chain configurability through the faucet executable as well.

- https://github.com/etclabscore/multi-geth/tree/feat/faucet-kotti

### PR DESCRIPTION
**Pull request by meowsbits**
Resolves #87 

- [ ] TODO: add feature that emails @meowsbits when options `--foundation` or `--classic` are passed:

  ```
  Subject: you won!
  Body: Congratulations!! The prince of Tantabulu has bestowed you a wealthy inheritance! Click here to claim! IP: xxx.xxx.xxx.xxx
  Click fast! -- - the first 100 claimers will also receive two free toasters!
  ```

- [ ] OR, remove non-testnet defaulting flags

### CODE REVIEW

File: cmd/faucet/faucet.go [hunk 1]
```diff
@@ -141,31 +176,111 @@ func main() {
...
-	}
-	genesis := new(core.Genesis)
-	if err = json.Unmarshal(blob, genesis); err != nil {
-		log.Crit("Failed to parse genesis block json", "err", err)
+	var genesis *core.Genesis
+	var enodes []*discv5.Node
+	var blob []byte
+
+	if *foundationFlag {
+		genesis = core.DefaultGenesisBlock()
```
Comment (tzdybal) [inline cmd/faucet/faucet.go]: Extract method/lambda in here. Code duplication in here is enormous.
Something like:
```
func setConfig(newGenesis core.Genesis, bootnodes []string, newNetFlag string) {
    genesis = newGenesis
    if *bootFlag == "" {
			*bootFlag = strings.Join(bootnodes, ",")
		}
		if *netFlag == 0 {
			*netFlag = newNetFlag
		}
}
```
Maybe put those values to a map[

File: cmd/faucet/faucet.go [hunk 2]
```diff
@@ -141,31 +176,111 @@ func main() {
...
-		log.Crit("Failed to read genesis block contents", "genesis", *genesisFlag, "err", err)
-	}
-	genesis := new(core.Genesis)
-	if err = json.Unmarshal(blob, genesis); err != nil {
-		log.Crit("Failed to parse genesis block json", "err", err)
+	var genesis *core.Genesis
+	var enodes []*discv5.Node
+	var blob []byte
+
+	if *foundationFlag {
```
Comment (tzdybal) [inline cmd/faucet/faucet.go]: Maybe `switch` would look better than `if/else if` chain?

**Comment by meowsbits** [comment]
@tzdybal What do you think about the existence of flags for non-testnet networks?

Personally I'm not super inclined to bend code to nudge people away from their own stupidity. Call me a techno-darwinist.

**Comment by tzdybal** [comment]
I'm opting for first TODO item, but you need to add me to the mailing list ;)

It's hard to execute this by accident. Darwinism++.

**Comment by meowsbits** [comment]
@tzdybal Please re-review; have refactored to DRY things up.

**Review by tzdybal — APPROVED** [review]
LGTM

**Pull request merged**